### PR TITLE
CRIMAPP-1119 Annualise and combine Other Income, and Other Benefits

### PR DESCRIPTION
**Pull request by ripan**
### **<u> Update amount of 'other' income payment</u> **

other income payment attributes:
- student_loan_grant
- board_from_family
- rent
- financial_support_with_access
- from_friends_relatives
- other

**When `other` income payment is present, update amount**
```
other.amount = ∑
- annualized(student_loan_grant)
- annualized(board_from_family)
- annualized(rent)
- annualized(financial_support_with_access)
- annualized(from_friends_relatives)
- annualized(other)
```

**When `other` income payment  is NOT present, create other income payment object and assign total amount to `other` payment_type**
```
other.amount = ∑
- annualized(student_loan_grant)
- annualized(board_from_family)
- annualized(rent)
- annualized(financial_support_with_access)
- annualized(from_friends_relatives)
```

Append 'other income payment attributes' to metadata details

### **<u> Update amount of 'other' income benefit</u> **

other income benefit attributes:
- jsa
- other

**When `other` income benefit is present, update amount**

```
other.amount = ∑
- annualized(jsa)
- annualized(other)
```

**When `other` income benefit  is NOT present, create other income benefit object and assign total amount to `other` payment_type**
```
other.amount = ∑
- annualized(jsa)
```

Append 'other income benefit attributes' to metadata details

mapping

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-1119

### CODE REVIEW

File: app/api/datastore/entities/v1/maat/income_details.rb [hunk 1]
```diff
@@ -13,8 +13,33 @@ class IncomeDetails < Grape::Entity
...
+          OTHER_INCOME_PAYMENTS = %w[
+            student_loan_grant
+            board_from_family
+            rent
+            financial_support_with_access
+          ].freeze
+
+          OTHER_INCOME_BENEFITS = %w[
+            jsa
+          ].freeze
```
Comment (hiboabd) [inline app/api/datastore/entities/v1/maat/income_details.rb:25]: Might be useful to add a comment here explaining why we have these are combined

Comment (hiboabd) [reply app/api/datastore/entities/v1/maat/income_details.rb:25]: Also not sure i follow why other is not included in these arrays

Comment (hiboabd) [reply app/api/datastore/entities/v1/maat/income_details.rb:25]: Also think `OTHER_INCOME_PAYMENTS`,  `OTHER_INCOME_BENEFITS` and `OtherPaymentCalculator` should be reworded as we have an other payment and benefit type so it gets confusing. Although I understand this mirrors the wording on maat

File: app/services/utils/annualized_amount_calculator.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module Utils
```
Comment (hiboabd) [inline app/services/utils/annualized_amount_calculator.rb:1]: Nice 👌

File: app/api/datastore/entities/v1/maat/income_details.rb (hunk 1 above)
Comment (ripan) [reply app/api/datastore/entities/v1/maat/income_details.rb]: @hiboabd I have refactored and followed SRP.

File: app/api/datastore/entities/v1/maat/income_details.rb [hunk 3]
```diff
@@ -13,8 +13,27 @@ class IncomeDetails < Grape::Entity
...
           def income_payments
-            object['income_payments'].reject { |p| p['payment_type'] == 'employment' }
+            Utils::MAAT::OtherIncomePaymentCalculator.new(
+              payments: object['income_payments'],
+            ).call
+          end
+
+          # TODO: : Need to confirm with MAAT team if they require the following income_benefit objects
+          # in response once annualized
+          # - jsa
```
Comment (RoseSAK) [inline app/api/datastore/entities/v1/maat/income_details.rb:32]: Same with this one

**Review by RoseSAK — APPROVED** [review]

**Pull request merged**